mitodl/mitxonline
# 734 - registration validate username against openedx

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/734

#### What's this PR do?
Restricts a new `mitxonline` user from registering with a username that is currently taken by an OpenEdx user.  If a new `mitxonline` user attempts to use a username which is already taken by an OpenEdx user, they will be unable to submit the user registration form and presented with an error message.

#### How should this be manually tested?

1. Create an OpenEdx user with a known username.
2. Register a new user through `mitxonline` and attempt to use the same username as in Step 1.
3. Verify that you are unable to create a new user with the username from Step 1, and that you receive an error message.

If you are running `mitxonline` and `openedx` locally, you can also test out the workflow when `mitxonline` cannot connect to `openedx` by only running `mitxonline` locally.  You should perform the same steps listed above, but you will not receive an error message and you should be allowed to successfully submit the registration form.

#### Where should the reviewer start?

- There is a new method added to the `openedx` API service which handles making the request to the `openedx` registration verification service.
- I have added a call to the `openedx` API service method to the User Serializer username validation.
- There is a custom exception that is thrown if the `openedx` API call is unsuccessful.

#### Screenshots (if appropriate)
![Screen Shot 2022-07-29 at 12 10 07 PM](https://user-images.githubusercontent.com/8311573/181800505-90d84df8-207c-4d26-a5c1-30729e00d0da.png)
